### PR TITLE
GB-6282 - Resume Windows support with Bun 1.1

### PR DESCRIPTION
**Pull request by yoav-lavi**
# Description

## Features

- Resumes Windows support via Bun 1.1.0 (NPM only for now)

## Tooling

- Upgrades Bun to 1.1.0
- Prevents bundling an unused Cloudflare dependency that breaks Windows builds and included build time env vars

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [ ] 🐛 Fix
- [x] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: engine/flake.nix [hunk 1]
```diff
@@ -12,7 +12,7 @@
   description = "Grafbase engine development environment";
 
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
```
Comment (yoav-lavi) [inline engine/flake.nix:15]: Will remove this when the PR is ready

File: flake.nix [hunk 2]
```diff
@@ -28,7 +28,7 @@
         flake-utils.follows = "flake-utils";
       };
     };
-    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable-small";
```
Comment (yoav-lavi) [inline flake.nix:31]: Will remove this when the PR is ready

**Review by tomhoule — APPROVED** [review]
lgtm!

File: cli/wrappers/bun.lockb
[the remark this answers is not in the document]
Comment (yoav-lavi) [reply cli/wrappers/bun.lockb:1]: Yeah, I switched the CI to use Bun to build this due to the added resolution

**Pull request merged**